ros-controls/ros_control
# Update CMakeLists.txt and package.xml

Various fixes and formatting for `CMakeLists.txt` and `package.xml` files, including fixes suggested by `catkin_lint`.

---
#### Apply consistent format to CMakeLists.txt and package.xml files

The format is largely based on the existing files. The most important change is that this PR corrects the order of commands in the `CMakeLists.txt`.

I'm not at all attached to the format used in the CMakeLists.txt, happy to gut out the comments or make other stylistic changes as desired. I would mostly just like them to be consistent, as it makes it much easier to identify issues and to maintain.

----

#### Beyond stylistic changes, the following changes were made:

`combined_robot_hw`:
 - Remove unnecessary `boost` library dependency

`combined_robot_hw_tests`:
 - Remove unnecessary `boost` library dependency
 - Add missing `${catkin_EXPORTED_TARGETS}` dependency
 - Gut the `catkin_package()` and `install()` commands
 - Move all targets into the `if(CATKIN_ENABLE_TESTING)` block
 - Update package description
 - Add missing dependencies to `package.xml`

`controller_interface`:
 - Remove `rosunit` dependency (See note at bottom)
 - Remove unneeded pluginlib dependency

`controller_manager`:
 - Add missing `roscpp` dependency
 - Add missing `${catkin_EXPORTED_TARGETS}` dependency
 - Add missing `exec_depends` for python scripts

`controller_manager_msgs`:
 - Add missing `exec_depends` for python scripts

`controller_manager_tests`:
 - Remove unnecessary `boost` include_dir dependency
 - Add missing `hardware_interface` and `roscpp` dependencies
 - Add missing `${catkin_EXPORTED_TARGETS}` dependency
 - Gut the `catkin_package()` and `install()` commands
 - Move all targets into the `if(CATKIN_ENABLE_TESTING)` block
 - Update package description
 - Update dependencies in `package.xml`

`hardware_interface`:
 - Add missing `find_package(Boost REQUIRED)` for header-only library dependency `ptr_container`
 - Add missing `boost` depend in package.xml
 - Remove redundant `rosconsole` dependency (Already included via `roscpp`)
 - Remove unused `rostest` test_depend
 - Remove `rosunit` dependency (See note at bottom)

`joint_limits_interface`:
 - Remove `liburdfdom-dev` dependency (See https://github.com/ros-controls/ros_control/pull/404#discussion_r365019612)

`rqt_controller_manager`:
 - Install python scripts with `catkin_install_python()` rather than the default cmake `install()`
 - Update package description
 - Add missing dependencies

`transmission_interface`:
 - Add missing `find_package(Boost REQUIRED)` for header-only library dependency `lexical_cast`
 - Add missing `boost` depend in package.xml
 - Remove unused `Boost` dependencies in tests
 - Fix `find_package()` in tests, to not override previous `find_package(catkin [...])`. See https://github.com/ros/rosdistro/issues/3010#issuecomment-42116904

---

#### Notes:

I was unable to find up-to-date documentation on `rosunit` and the corresponding `catkin_add_gtest`/`catkin_add_gmock`. In particular, I don't know if it is necessary to `find_package(rosunit)` or `<test_depend>rosunit</test_depend>`. I believe the `find_package` is unecessary. I think the `test_depend` is required, but I'm not sure.

For header-only Boost libraries, I used `find_package(Boost REQUIRED)` and `include_directories(${Boost_INCLUDE_DIRS})`. I was not able to find any other examples of this use in large ROS packages, so I just wanted to highlight it here in case there's a better approach.

---

#### TODO:

Couple things I just want to look through before changing this from a draft to a normal PR:

 - ~Make sure `add_dependencies(${catkin_EXPORTED_TARGETS})` is used everywhere it's required~ Done in fa93166
 - ~Make sure `roscpp` is depended upon directly everywhere that it's required~ Done in 79b504c
 - ~Investigate the `urdf`/`liburdf-dev` situation, and see if things can be simplified~ Done in 7ca25ab
